07week/main.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+console.log("loading script.js file")
+
+window.onload = function() {
+    console.log("window loaded")
+    getPosts()
+}
+
+let getPosts = function() {
+    console.log("Inside the post method, about to make a fetch request")
+    let fetchPromise = fetch('https://randomuser.me/api/?results=10')
+
+    let dataPromise = fetchPromise.then(function(response) {
+        console.log("The packaging of the amazon item has arrived", response)
+        return response.json()
+    })
+
+    dataPromise.then(function(data) {
+        console.log("We have opened the packaging and the item/data is here! data =", data)
+        processContact(data.results)
+    })
+    console.log("Request sent off...")
+}
+
+function processContact(contacts) {
+    console.log("the contacts are =", contacts)
+    contacts.forEach(function(person) {
+
+            console.log("Updating the HTML for posts = ", contacts, "this is inside the for each =", person)
+            let postsUl = document.getElementById("posts")
+            let postLi = document.createElement("li")
+
+            postLi.innerText = person.name.first + " " + person.name.last
+            postsUl.appendChild(postLi)
+        
+            let userPic = document.createElement("img")
+            console.log(userPic)
+            userPic.setAttribute("src", person.picture.thumbnail)
+            postsUl.appendChild(userPic)
+
+            document.getElementById("info").addEventListener('click', function () {
+            let ulInfo = document.getElementById("info")
```

Comment:
this is adding 1 info button that is showing all the contacts at the same time. 
Wold have been better if you created a new info button per contact and had that show the info for one given contact at a time.